src/plugins/data/server/search/search_service.ts

```diff
@@ -130,7 +130,8 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
         this.initializerContext.config.legacy.globalConfig$,
         this.logger,
         usage,
-        dataSource
+        dataSource,
+        core.opensearch
```

Comment:
can we get config value from `this.initializerContext.config.legacy.globalConfig$` instead of adding new parameter? What globalConfig returns?

Reply:
Thanks for the comment~

**To answer the question:**

"hosts" is not exposed in global config.
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/790c076fd4c8f38ade5482cede6d92d10792be54/src/core/server/plugins/types.ts#L291

**More Context:**

I used to think of exposing "hosts"global config to avoid this parameter and that change is in this PR's first commit.
But get feedbacks from folks, it will be risky to export a new attribute into global config.

So this parameter is needed.

Reply:
Got it. Thank you for looking into it.